spec/grid_spec.rb

```diff
@@ -111,13 +135,49 @@ def set_up_sunk_submarine
 
   it "returns false if not all ships are sunk" do
     sunk_submarine = set_up_sunk_submarine
-    grid.add_sunk_ship(sunk_submarine)
+    grid.register_sunk_ship_positions(sunk_submarine)
     submarine2 = Ship.new("submarine", 1)
-    grid.mark_ship_positions(2, "G", submarine2, :vertical)
+    submarine2.register_position(2, "G", :vertical)
+    grid.add_ship(submarine2)
 
     verdict = grid.end_game?
 
     expect(verdict).to eq(false)
   end
 
+  describe "returns result :hit, :sunk or :water after each attack" do
+    it "returns :hit for hit occupied cell" do
+      ship.register_position(1, "A", :horizontal)
+      grid.add_ship(ship)
+
+      attacked_cell = [1, "A"]
+
+      expect(grid.show_result(attacked_cell)).to eq(:hit)
+      expect(ship.cells_hit.include?(attacked_cell)).to eq(true)
+    end
+
+    it "returns :water for no hit cell" do
+      ship.register_position(1, "A", :horizontal)
+      grid.add_ship(ship)
+
+      attacked_cell = [8, "A"]
+
+      expect(grid.show_result(attacked_cell)).to eq(:water)
+      expect(ship.cells_hit.include?(attacked_cell)).to eq(false)
+    end
+
+    it "returns :sunk for all cells of ship hit" do
+      ship.register_position(1, "A", :horizontal)
+      grid.add_ship(ship)
+
+      attacked_cell = [1, "A"]
+      attacked_cell2 = [2, "A"]
+      ship.register_cells_hit(attacked_cell)
+      ship.register_cells_hit(attacked_cell2)
+
+      expect(grid.show_result(attacked_cell)).to eq(:sunk)
+      expect(grid.ships_sunk_positions).to eq([[1, "A"], [2, "A"]])
```

Comment:
I think you have too many expectations in this test. Because what you really want to test is that it returns :sunk, don't you? At least that's what the test name says. So this test might fail for the wrong reasons if it fails for the expectation in line 179... I would only have one expectation in a test.